sugoi-api-file-store-provider/src/main/java/fr/insee/sugoi/store/file/FileReaderStore.java

```diff
@@ -13,82 +13,198 @@
 */
 package fr.insee.sugoi.store.file;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import fr.insee.sugoi.core.model.PageResult;
 import fr.insee.sugoi.core.model.PageableResult;
 import fr.insee.sugoi.core.store.ReaderStore;
 import fr.insee.sugoi.model.Application;
 import fr.insee.sugoi.model.Group;
 import fr.insee.sugoi.model.Organization;
 import fr.insee.sugoi.model.User;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.NotImplementedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
 
 public class FileReaderStore implements ReaderStore {
 
-  public FileReaderStore(Map<String, String> generateConfig) {}
+  @Autowired ResourceLoader resourceLoader;
+  private Map<String, String> config;
+  private ObjectMapper mapper = new ObjectMapper();
 
-  public FileReaderStore() {}
+  public FileReaderStore(Map<String, String> generateConfig) {
+    this.config = generateConfig;
+  }
 
   @Override
   public User getUser(String id) {
-    // TODO Auto-generated method stub
+    Resource realmsResource = resourceLoader.getResource(config.get("user_source") + id);
+    if (realmsResource.exists()) {
+      User user = loadResourceContent(realmsResource, User.class);
+      // suborganization is loaded as an independant resource
+      if (user.getOrganization() != null && user.getOrganization().getIdentifiant() != null) {
+        user.setOrganization(getOrganization(user.getOrganization().getIdentifiant()));
```

Comment:
We may want to catch the exception in case the organization can't be loaded, to return an user anyway (even without org), what do you think ?

Reply:
Can you take a look at last commit to see if it matches what you suggested ?

Reply:
yes, seems good, thx